mantidproject/mantid
# Use native file browsers on GTK-based linux environments

**Description of work**
Install an extra pacakge from conda-forge called qt-gtk-platformtheme. This package provides the GTK+ 3 platform theme, which makes Qt use native icon themes, fonts, and dialogs on GTK-based environments.

**Purpose of work**
Since moving to Conda, the file browsers on IDAaaS have become unusable when navigating to large directories. This is because the native browsers weren't being used. Installing this extra package solves the issue.

**Further detail of work**
Had to change the occt pinning in the conda recipes in order to resolve the dependency chain. It is now consistent with what was already in the developer environments (which is what is used to run all of our tests!).

**To test:**
(On linux only) Create a new conda environment, activate it, and install mantidworkbench:
```
conda create -n mantid-gtk-test
conca activate mantid-gtk-test
mamba install -c "thomashampson/label/gtk-test" mantidworkbench
```
Then launch with `workbench` and open up a file browser and navigate to a big directory. E.g. this one was causing problems on IDAaaS:
`/archive/NDXLARMOR/Instrument/data/cycle_22_3/`
The GUI should now remain responsive while the file browser is populating.

I have tested myself on IDAaaS by creating a new conda env and it is working well.

**Optionally** you can update your development conda environment after checking out this branch with
`mamba env update -f mantid-developer-linux.yaml --prune`
which will install the new package. Then when you launch your developer build of workbench you should see the gtk theme file browser.

Fixes #34820



---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
